src/js/components/homepage/visualizations/treemap/BudgetSubfunctions/BudgetSubfunctionsMap.jsx

```diff
@@ -0,0 +1,237 @@
+/**
+ * BudgetSubfunctionsMap.jsx
+ * Created by Emily Gullo 03/15/2017
+ **/
+
+import React from 'react';
+import { hierarchy, treemap, treemapBinary, treemapSlice } from 'd3-hierarchy';
+import _ from 'lodash';
+import * as MoneyFormatter from 'helpers/moneyFormatter';
+
+import BudgetFunctionCell from '../BudgetFunctions/BudgetFunctionCell';
+import TreeMapTooltip from '../TreeMapTooltip';
+
+const propTypes = {
+    category: React.PropTypes.object,
+    colors: React.PropTypes.array,
+    subfunction: React.PropTypes.object,
+    tooltipStyles: React.PropTypes.object
+};
+
+export default class BudgetSubfunctionsMap extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            windowWidth: 0,
+            visualizationWidth: 0,
+            visualizationHeight: 286,
+            finalNodes: [],
+            hoveredFunction: -1,
+            negativeSubfunctions: []
+        };
+
+        this.handleWindowResize = _.throttle(this.handleWindowResize.bind(this), 50);
+        this.buildTree = this.buildTree.bind(this);
+        this.createTooltip = this.createTooltip.bind(this);
+        this.toggleTooltipIn = this.toggleTooltipIn.bind(this);
+        this.toggleTooltipOut = this.toggleTooltipOut.bind(this);
+    }
+
+    componentDidMount() {
+        this.handleWindowResize();
+        window.addEventListener('resize', this.handleWindowResize);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.subfunction.children.length > 0) {
```

Comment:
Can we restrict this `if` statement more? Right now it will rebuild the tree any time any prop changes as long as the subfunction children are populated (which will be any time after the initial JSON is loaded)